README.md

```diff
@@ -1,59 +1,94 @@
-# Getting started with thread statistics on Mbed OS
+![](./resources/official_armmbed_example_badge.png)
+# Thread Statistics Mbed OS Example
 
-This guide reviews the steps required to get thread statistics on Mbed OS platform.
+This guide reviews the steps required to get thread statistics on an Mbed OS enabled platform.
 
-Please install [mbed CLI](https://github.com/ARMmbed/mbed-cli#installing-mbed-cli).
+You can build this project with all supported [Mbed OS build tools](https://os.mbed.com/docs/mbed-os/latest/tools/index.html). However, this example project specifically refers to the command-line interface tool [Arm Mbed CLI](https://github.com/ARMmbed/mbed-cli#installing-mbed-cli).
 
-## Import the example application
+## Application functionality
 
-From the command-line, import the example:
+The `main()` function outputs on the serial interface about a maximum of 8 threads information like `id`, `name`, `state`, `priority`, `stack_space`, `stack_size`.
 
-```
-mbed import mbed-os-example-thread-statistics
-cd mbed-os-example-thread-statistics
-```
+## Preqrequisites
 
-### Now compile
+1. [Install Mbed CLI](https://os.mbed.com/docs/mbed-os/latest/tools/installation-and-setup.html).
+1. Determine which toolchain supports your target.
 
-Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler 5:
+   Depending on the target, you can build the example project with the GCC_ARM, ARM or IAR toolchain. To learn which toolchain supports your target, run this command:
+
+   ```bash
+   $ mbed compile -S
+   ```
+
+1. Clone this repository on your system.
+1. Change the current directory to where the project was cloned.
+
+## Building and running
+
+1. Connect a USB cable between the USB port on the target and the host computer.
+1. Run this command to build the example project and program the microcontroller flash memory:
+
+    ```bash
+    $ mbed compile -m <TARGET> -t <TOOLCHAIN> --flash --sterm
+    ```
+
+(Note: You can use the Mbed CLI command-line option "--sterm" to open a serial terminal after flashing.)
+
+Your PC may take a few minutes to compile your code.
+
+The binary is located at `./BUILD/<TARGET>/<TOOLCHAIN>/mbed-os-example-thread-statistics.bin`.
+
+Alternatively, you can manually copy the binary to the target, which gets mounted on the host computer through USB.
+
+## Expected output
+
+The serial terminal shows an output similar to the following:
 
 ```
-mbed compile -m K64F -t ARM
-```
+--- Terminal on /dev/ttyACM0 - 9600,8,N,1 ---
+ID: 0x20000e08 
+kName: main 
+State: 2 
+Priority: 24 
+Stack Size: 4096 
+Stack Space: 3944 
+
+ID: 0x20000f58 
+Name: rtx_idle 
+State: 1 
+Priority: 1 
+Stack Size: 512 
+Stack Space: 448 
 
-Your PC may take a few minutes to compile your code. At the end, you see the following result:
+ID: 0x20000f14 
+Name: rtx_timer 
+State: 3 
+Priority: 40 
+Stack Size: 768 
+Stack Space: 664
+```
```

Comment:
In the example, could you start a few threads so we see more statistics (perhaps threads in the other states)?